File: cmd/api/dto/item_dto.go
package dto

import (
	"time"

	"github.com/google/uuid"
)

type AddItemInputDTO struct {
	ParentID      uuid.UUID `json:"parentId"`
	Code          string    `json:"code"`
	Name          string    `json:"name"`
	SalesName     string    `json:"salesName"`
	Description   string    `json:"description"`
	Uom           string    `json:"uom"`
	AnvisaCode    string    `json:"anvisaCode"`
	AnvisaDueDate time.Time `json:"anvisaDueDate"`
	SupplierCode  string    `json:"supplierCode"`
	Cst           string    `json:"cst"`
	SusCode       string    `json:"susCode"`
	NcmCode       string    `json:"ncmCode"`
}

type ItemOutputDTO struct {
	ID            uuid.UUID `json:"id"`
	ParentID      uuid.UUID `json:"parentId"`
	Code          string    `json:"code"`
	Name          string    `json:"name"`
	SalesName     string    `json:"salesName"`
	Description   string    `json:"description"`
	Uom           string    `json:"uom"`
	AnvisaCode    string    `json:"anvisaCode"`
	AnvisaDueDate time.Time `json:"anvisaDueDate"`
	SupplierCode  string    `json:"supplierCode"`
	Cst           string    `json:"cst"`
	SusCode       string    `json:"susCode"`
	NcmCode       string    `json:"ncmCode"`
	CreatedBy     uuid.UUID `json:"createdBy"`
	CreatedAt     time.Time `json:"createdAt"`
	UpdatedBy     uuid.UUID `json:"updatedBy"`
	UpdatedAt     time.Time `json:"updatedAt"`
}

type UpdateItemInputDTO struct {
	ID            uuid.UUID `json:"id"`
	ParentID      uuid.UUID `json:"parentId"`
	Code          string    `json:"code"`
	Name          string    `json:"name"`
	SalesName     string    `json:"salesName"`
	Description   string    `json:"description"`
	Uom           string    `json:"uom"`
	AnvisaCode    string    `json:"anvisaCode"`
	AnvisaDueDate time.Time `json:"anvisaDueDate"`
	SupplierCode  string    `json:"supplierCode"`
	Cst           string    `json:"cst"`
	SusCode       string    `json:"susCode"`
	NcmCode       string    `json:"ncmCode"`
}

File: src/Omini.Opme.Be.Api/Dtos/ItemDto.cs
namespace Omini.Opme.Be.Api;

public sealed record ItemCreateDto
{
    public Guid Id { get; set; }
    public string Code { get; set; }
    public string Name { get; set; }
    public string SalesName { get; set; }
    public string Description { get; set; }
    public string Uom { get; set; }
    public string AnvisaCode { get; set; }
    public DateTime AnvisaDueDate { get; set; }
    public string SupplierCode { get; set; }
    public string Cst { get; set; }
    public string SusCode { get; set; }
    public string NcmCode { get; set; }
}


public sealed record ItemOutputDto
{
    public Guid Id { get; set; }
    public string Code { get; set; }
    public string Name { get; set; }
    public string SalesName { get; set; }
    public string Description { get; set; }
    public string Uom { get; set; }
    public string AnvisaCode { get; set; }
    public DateTime AnvisaDueDate { get; set; }
    public string SupplierCode { get; set; }
    public string Cst { get; set; }
    public string SusCode { get; set; }
    public string NcmCode { get; set; }
}